elasticsearch-transport/lib/elasticsearch/transport/transport/sniffer.rb

```diff
@@ -30,20 +31,26 @@ def initialize(transport)
         def hosts
           Timeout::timeout(timeout, SnifferTimeoutError) do
             nodes = transport.perform_request('GET', '_nodes/http').body
-            hosts = nodes['nodes'].map do |id,info|
-              addr_str = info["#{transport.protocol}_address"].to_s
-              matches = addr_str.match(ES1_RE_URL) || addr_str.match(ES2_RE_URL)
-              if matches
-                host = matches[1].empty? ? matches[2] : matches[1]
-                port = matches[3]
-                info.merge :host => host, :port => port, :id => id
-              end
-            end.compact
-
-            hosts.shuffle! if transport.options[:randomize_hosts]
-            hosts
+            hosts_from_nodes(nodes)
           end
         end
+
+        protected
+
+        def hosts_from_nodes(nodes)
```

Comment:
Since we're adding a method on the public API, can we have documentation for this method? Also, can we have unit test specifically for this method, or do we rely on testing it in other parts of the code?